deploy-board/deploy_board/webapp/group_view.py

```diff
@@ -181,33 +181,28 @@ def update_launch_config(request, group_name):
         else:
             launchRequest["assignPublicIp"] = False
 
-        userPassInSpiffeID = None
+        desired_spiffe_id = None
         passInUserDataList = params["userData"].splitlines()
         for item in passInUserDataList:
-                if ":" in item:
-                    pair = item.split(":")
-                    if pair[0] == "spiffe_id":
-                        userPassInSpiffeID = pair[1]
-                        break
+            desired = item.split(":", 1)
+            if desired[0] == "spiffe_id":
+                desired_spiffe_id = desired[1]
+                break
 
+        actual_spiffe_id = None
         group_info = autoscaling_groups_helper.get_group_info(request, group_name)
         launch_config = group_info.get("launchInfo")
         if 'userData' in launch_config:
             userDataList = launch_config['userData'].splitlines()
             for item in userDataList:
-                if ":" in item:
-                    pair = item.split(":")
-                    if pair[0] == "spiffe_id" and userPassInSpiffeID is None:
-                        log.error("Teletraan does not support user to remove spiffe_id in userData field")
-                        raise TeletraanException("Teletraan does not support user to remove spiffe_id")
-
-                    if pair[0] == "spiffe_id" and pair[1] != userPassInSpiffeID:
-                        log.error("Teletraan does not support user to update spiffe_id in userData field")
-                        raise TeletraanException("Teletraan does not support user to update spiffe_id")
-
-                    if pair[0] != "spiffe_id" and userPassInSpiffeID is not None:
-                        log.error("Teletraan does not support user to create spiffe_id in userData field")
-                        raise TeletraanException("Teletraan does not support user to create spiffe_id")
+                actual = item.split(":", 1)
```

Comment:
should we make this into a function like "getSpiffe_id" since it is being dup from previous code